lib/sync_engine/abstract/user_manager.rb

```diff
@@ -46,18 +50,18 @@ def success_auth_response(_user, _params)
       raise Other::NotImplementedError 'Must override'
     end
 
-    def registration_params(params)
-      params.permit(
-        :pw_func,
-        :pw_alg,
-        :pw_cost,
-        :pw_key_size,
-        :pw_nonce,
-        :pw_salt,
-        :origination,
-        :created,
-        :version
-      )
+    def registration_params(params, with_defaults = true)
+      defaults = {}
+
+      registration_fields.each do |field|
```

Comment:
Shouldn't this be inside the `if with_defaults` clause? Same with line 54. Then line 64 should be in an `else` clause so that it's easy to follow.